frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -326,7 +348,10 @@ fn decl_outer_event<'a>(
 					);
 					return Err(syn::Error::new(module_declaration.name.span(), msg));
 				}
-				let tokens = quote!(#module #instance #generics ,);
+
+				let index = module_declaration.index.expect("All index have been assigned");
```

Comment:
Maybe we should create a function `index` that does this expect internally? :D

Reply:
Yes but I didn't wanted to iterate on all module everytime.
Actually i was thinking adding a structure Module, which will contains the index, and convert all ModuleDeclaration to it.
This would be cleaner I can do if wanted.

Reply:
Sounds good to me:)

Reply:
done here https://github.com/paritytech/substrate/pull/6969/commits/61019e704f211ef400db9262d99fbf635dba7997